dds/idl/dds_generator.h

```diff
@@ -409,6 +409,8 @@ namespace AstTypeClassification {
         ((dynamic_cast<AST_String*>(type)->max_size()->ev()->u.ulval == 0)
         ? 0 : CL_BOUNDED) |
         ((type->node_type() == AST_Decl::NT_wstring) ? CL_WIDE : 0);
+    case AST_Decl::NT_map:
```

Comment:
To stop the errors, you can test if TAO has maps by updating this:

https://github.com/objectcomputing/OpenDDS/blob/51e2ed25da43f5a52ac1189756edffdc2e049482/dds/DCPS/Definitions.h#L69-L75

to match the changes you made to `idl_features.h` in TAO, then you should be able to use:

```suggestion
#if OPENDDS_HAS_MAP
    case AST_Decl::NT_map:
```
For IDL you can use the TAO_IDL macros directly like in explicit ints test:

https://github.com/objectcomputing/OpenDDS/blob/51e2ed25da43f5a52ac1189756edffdc2e049482/tests/DCPS/Compiler/explicit_ints/test.idl#L1-L3